lib/propolis/src/accessors.rs

```diff
@@ -368,13 +402,13 @@ fn print_basic(match_node: Option<NodeKey>) -> impl Fn(PrintNode) {
 
 type TreeBackref<T> = Arc<Mutex<Tree<T>>>;
 
-struct NodeEntry<T> {
+struct NodeEntry<T: AccessedResource> {
     tree: TreeBackref<T>,
-    resource: Option<Arc<T>>,
+    res_leaf: Option<T::Leaf>,
```

Comment:
it feels worth noting explicitly that we'll be OK with `res_leaf` differing in nodes across a tree (between `None` and, roughly, `Some(node.tree.res_root)`). mostly interesting that `res_leaf` is now more of a cache that is flushed when tree is adopted or the root resource is changed.

Reply:
I've added some color here, which I hope is adequate?